zookeeper-server/src/main/java/org/apache/zookeeper/server/PrepRequestProcessor.java

```diff
@@ -1154,4 +1158,10 @@ private void setTxnDigest(Request request, PrecalculatedDigest preCalculatedDige
         }
         request.setTxnDigest(new TxnDigest(digestCalculator.getDigestVersion(), preCalculatedDigest.treeDigest));
     }
+
+    private static void overwriteAclListToCreatorId(List<ACL> aclList, Id creatorId) {
```

Comment:
As discussed in the review meeting, it would be great if we could do some sanity checks:
in a high throughput/write qps situation:
- are we doing any unnecessary loops? (time complexity)
- are we creating objects that are not needed? (heap usage)
- are static config reads optimized (eg.. no repeated string parsing or boolean checks, etc.)

Great job!